File: backend/src/main/java/com/ems/backend/service/SalaryService.java
package com.ems.backend.service;

import com.ems.backend.dto.Response;
import com.ems.backend.dto.SalaryDto;

public interface SalaryService {

    Response addSalary(SalaryDto salaryDto);

    Response getAllSalaries();

    Response updateSalary(Long id , SalaryDto salaryDto);

    Response deleteSalary(Long id);

    Response getSalaryById(Long id);

    Response filterSalaries(String month , Long departmentId , Long employeeId );

    Response searchSalaries(String keyword);

}

File: frontend/src/services/salaryService.js
import api from "./api";

const API = "http://localhost:8080/api/salaries";

export const getAllSalaries = () => api.get(API);

export const addSalary = (salaryData) => api.post(API,salaryData);

export const updateSalary = (id , salaryData) =>api.put(`${API}/${id}`,salaryData);

export const deleteSalary = (id) => api.delete(`${API}/${id}`);

export const getSalaryById = (id) => api.get(`${API}/${id}`);

export const filterSalaries = (month, departmentId, employeeId) => {
  const params = new URLSearchParams();

  if (month) params.append("month", month);
  if (departmentId) params.append("departmentId", departmentId);
  if (employeeId) params.append("employeeId", employeeId);

  return api.get(`${API}/filter?${params.toString()}`);
};

export const searchSalaries = (keyword) =>
  api.get(`${API}/search`, {
    params: { keyword },
  });
